indefinitepagerindicator/src/main/java/com/rbrooks/indefinitepagerindicator/IndefinitePagerIndicator.kt

```diff
@@ -195,11 +195,11 @@ class IndefinitePagerIndicator @JvmOverloads constructor(context: Context, attrs
         viewPager?.removeOnPageChangeListener(this)
         viewPager = null
 
-        this.recyclerView?.removeOnScrollListener(internalRecyclerScrollListener)
+        this.recyclerView?.removeOnScrollListener(internalRecyclerScrollListener!!)
```

Comment:
I really hate the use of `!!` even though this is a safe call. In the case of removeOnScrollListener, nothing will occur if the passed internalRecyclerScrollListener is null due to a null check.